### DISCUSSION TITLE
ShouldQuote depending on field value?

### DISCUSSION THREAD

**Discussion by rorourke-iot** [JoshClose/CsvHelper#2259] [Q&A] [answered]
My investigation of the values in an instance of `ShouldQuoteArgs` leads me to believe the answer to my question is "no". That said, I'm asking anyway in case I'm missing something regarding getting to the value of a particular field I'm evaluating within a `ShouldQuote` configuration.

Basically, looking to quote empty fields (along with any other field value which would normally get quoted).

**Comment by JoshClose** [comment, marked as answer]
```cs
var config = new CsvConfiguration(CultureInfo.InvariantCulture)
{
	ShouldQuote = args =>
	{
		if (args.Field.Length == 0)
		{
			return true;
		}
		
		return ConfigurationFunctions.ShouldQuote(args);
	}
};
```

**Reply by rorourke-iot** [reply]
Excellent. Thanks @JoshClose. My investigation wasn't sufficient. :(

I dumped all the `args` and only looked at the first set thinking `Field` was always the name of the field being written (i.e., the header). That said, I still would've needed the `ConfigurationFunctions` info, I think.

**Reply by JoshClose** [reply]
Those are the defaults used for configuration. I put them separately like that so you can use them if you need to when creating your own.

**Reply by JoshClose** [reply]
...I don't think they're documented anywhere.
